runtime/oti/j9port.h

```diff
@@ -43,6 +43,10 @@
 #include <unistd.h>
 #endif
 
+#if defined(OMR_MUSL_CLIB)
+#include <pthread.h>
```

Comment:
Where are other platforms getting the definition of `typedef pthread_spinlock_t spinlock_t;` from?  I'd expect MUSL to pick up pthread in the same way other platforms are.

Reply:
@DanHeidinga  for other platforms its getting from `sys/types.h` which includes `bits/pthreadtypes.h` which has `pthread_spinlock_t`. Any of the standard header file included by this file might have included `sys/types.h`.

In case of alpine `pthread_spinlock_t` is an optional definition which is only specified if defined `__NEED_pthread_spinlock_t`. `pthread_spinlock_t` is defined in `bits/alltypes.h` guarded with `__NEED_pthread_spinlock_t` macro

**code from musl - bits/alltypes.h:**
```
#if defined(__NEED_pthread_spinlock_t) && !defined(__DEFINED_pthread_spinlock_t)
typedef int pthread_spinlock_t;
#define __DEFINED_pthread_spinlock_t
#endif
```

`pthread.h` in MUSL defines `__NEED_pthread_spinlock_t` and includes `bits/alltypes.h` so we could get the `pthread_spinlock_t` definition.

**code from musl - pthread.h:**
```
#define __NEED_time_t
#define __NEED_clockid_t
#define __NEED_struct_timespec
#define __NEED_sigset_t
#define __NEED_pthread_t
#define __NEED_pthread_attr_t
#define __NEED_pthread_mutexattr_t
#define __NEED_pthread_condattr_t
#define __NEED_pthread_rwlockattr_t
#define __NEED_pthread_barrierattr_t
#define __NEED_pthread_mutex_t
#define __NEED_pthread_cond_t
#define __NEED_pthread_rwlock_t
#define __NEED_pthread_barrier_t
+-----------------------------------+
| #define __NEED_pthread_spinlock_t |
+-----------------------------------+
#define __NEED_pthread_key_t
#define __NEED_pthread_once_t
#define __NEED_size_t
+----------------------------+
| #include <bits/alltypes.h> |
+----------------------------+
```

Reply:
Thanks for the clarification @bharathappali 

@ashu-mehra's suggestion to include pthread based on `defined(LINUX) && !defined(J9ZTPF)` is a more explicit change and is clear for all platforms.  I think that's the way we should proceed.

Reply:
Sure @DanHeidinga Will update the PR accordingly